cpp/lib/src/Zotero.cc

```diff
@@ -160,6 +160,36 @@ bool Web(const Url &zts_server_url, const TimeLimit &time_limit, Downloader::Par
 } // namespace TranslationServer
 
 
+template <typename T>
+void VisitJsonMetadataNodes(const std::string &node_name, const std::shared_ptr<JSON::JSONNode> &node,
+                            const SiteParams &site_params, const T callback)
+{
+    switch (node->getType()) {
+    case JSON::JSONNode::OBJECT_NODE:
+        for (const auto &key_and_node : *JSON::JSONNode::CastToObjectNodeOrDie(node_name, node))
```

Comment:
You could just cast "node" to an object node here and a few lines below w/o using our safe and slower CastToObjectNodeOrDie.